src/sema/functions.rs

```diff
@@ -192,6 +192,14 @@ pub fn contract_function(
                     ));
                     success = false;
                 } else {
+                    if ns.target == Target::Solana {
+                        ns.diagnostics.push(Diagnostic::warning(
+                            *loc,
+                            "Selector (or discriminator) overriding does not work on Solana. \
+                              The IDL spec needs updating to make this feature available."
```

Comment:
This error is not helpful. What needs updating? What is the user supposed to do? 

How about `selector overriding may not have the desired affect. The metadata file does not contain the overriden selector (known as the discriminator) so the contract will be unusable.`

Reply:
Are we turning this into an error instead of a warning?

Reply:
I should have said warning. The `solang idl` command produces selector overrides.